src/NumericalAlgorithms/LinearSolver/Tags.hpp

```diff
@@ -0,0 +1,89 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines DataBox tags for the linear solver
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "NumericalAlgorithms/LinearSolver/IterationId.hpp"
+
+/*!
+ * \ingroup LinearSolverGroup
+ * \brief Functionality for solving linear systems of equations
+ */
+namespace LinearSolver {
+namespace Tags {
+
+/*!
+ * \ingroup LinearSolverGroup
+ * \brief Holds an `IterationId` that identifies a step in the linear solver
+ * algorithm
+ */
+struct IterationId : db::SimpleTag {
+  static std::string name() noexcept { return "IterationId"; }
+  using type = LinearSolver::IterationId;
+};
+
+/*!
+ * \ingroup LinearSolverGroup
+ * \brief The operand that the local linear operator \f$A\f$ is applied to
+ *
+ * \details The result of the operation should be wrapped in
+ * `LinearSolver::Tags::Projection`.
+ */
+template <typename Tag>
+struct Operand : db::PrefixTag, db::SimpleTag {
+  static std::string name() noexcept {
+    return "LinearOperand(" + Tag::name() + ")";
+  }
+  using type = typename Tag::type;
+  using tag = Tag;
+  static constexpr bool should_be_sliced_to_boundary = true;
+};
+
+/*!
+ * \ingroup LinearSolverGroup
+ * \brief The projection of the linear operator \f$A\f$ onto `Tag`
+ */
+template <typename Tag>
+struct Projection : db::PrefixTag, db::SimpleTag {
+  static std::string name() noexcept {
+    return "LinearProjection(" + Tag::name() + ")";
+  }
+  using type = typename Tag::type;
+  using tag = Tag;
+};
+
+/*!
+ * \ingroup LinearSolverGroup
+ * \brief The residual \f$r=b - Ax\f$
```

Comment:
Where does `b` come from in all of this?

Reply:
If you mean where is it defined: In the documentation where I write $Ax=b$ is the linear equation to be solved.
If you mean how is it computed: In the element initializations from the problem at hand.

Reply:
I mean where is it stored or otherwise obtained from when setting this tag.

Reply:
During initialization it’s computed and passed to the linear solver init action. It typically doesn’t need to be stored, since the residual is updated iteratively.

Reply:
Oh and where it's handled it's represented by the existing `Tags::Source` prefix.